sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreCatalogSuite.scala

```diff
@@ -358,4 +361,23 @@ class DataSourceWithHiveMetastoreCatalogSuite
         Seq(table("src").count().toString))
     }
   }
+
+  test("SPARK-29869: fix HiveMetastoreCatalog#convertToLogicalRelation throws AssertionError") {
+    withTempPath(dir => {
+      val baseDir = s"${dir.getCanonicalFile.toURI.toString}/test"
+      val partitionLikeDir = s"${dir.getCanonicalFile.toURI.toString}/test/dt=20191113"
+      spark.range(3).selectExpr("id").write.parquet(partitionLikeDir)
+      withTable("non_partition_table") {
+        withSQLConf(HiveUtils.CONVERT_METASTORE_PARQUET.key -> "true") {
+          spark.sql(
+            s"""
+               |CREATE TABLE non_partition_table (id bigint)
```

Comment:
Isn't it a malformed table? Does hive ignore the directories for non-partitioned tables?

Reply:
Seems Hive return none when query this table(1.2.1):
> hive> select * from `xxxxx`.`xxxx`;
OK
Time taken: 25.301 seconds

But no assertion error

Reply:
Maybe you are right. Actually the data file path of this table is `/path/tablename/dt=yyyymmdd/dt=yyyymmdd/xxx.parquet`. I guess Hive does not recursively lookup load the data. So it return empty but not error.
And I found if when enable recursively lookup by `.option("recursiveFileLookup", true)`, the inferPartitioning will be disable. So `dt=yyyymmdd` won't be treated as partitionSpec.

So should I revert the code changes and only keep the assert detail information? Or auto rollback to do not use built-in Parquet reader to read?